service-coordtransform/src/main/java/fi/nls/oskari/spring/CoordinateTransformationAsyncController.java

```diff
@@ -0,0 +1,226 @@
+package fi.nls.oskari.spring;
+
+import com.fasterxml.jackson.core.JsonFactory;
+import com.fasterxml.jackson.core.JsonGenerator;
+import com.vividsolutions.jts.geom.Coordinate;
+import fi.nls.oskari.control.*;
+import fi.nls.oskari.log.LogFactory;
+import fi.nls.oskari.log.Logger;
+import fi.nls.oskari.service.OskariComponentManager;
+import fi.nls.oskari.spring.extension.OskariParam;
+import fi.nls.oskari.util.IOHelper;
+import fi.nls.paikkatietoikkuna.coordtransform.*;
+import org.springframework.stereotype.Controller;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.ResponseBody;
+import org.springframework.web.context.request.async.DeferredResult;
+
+import javax.servlet.http.HttpServletResponse;
+import java.io.*;
+import java.util.List;
+
+
+/**
+ * Handles long polling requests for coordinate transformation.
+ * Timeouts with the job id, or returns transformation result.
+ */
+@Controller
+public class CoordinateTransformationAsyncController {
+    protected static final Logger log = LogFactory.getLogger(CoordinateTransformationAsyncController.class);
+
+    protected static final String RESPONSE_COORDINATES = "resultCoordinates";
+    protected static final String RESPONSE_DIMENSION = "dimension";
+    protected static final String RESPONSE_JOB_ID = "jobId";
+
+    protected static final String FILE_EXT = "txt";
+    protected static final String FILE_TYPE = "text/plain";
+
+    private static long POLLING_TIMEOUT_MS = 45000l;
```

Comment:
Should this be 45000?

Reply:
The last char is lowercase L for long. True, it's not required and could be left out for clarity.

Reply:
Oops. my bad. I'm more accustomed to `45000L` (although for floats I use lowercase f, weird.)